src/api/route-services/samples.js

```diff
@@ -201,6 +176,50 @@ class SamplesService {
 
     return OK();
   }
+
+  getS3UploadLink(projectUuid, sampleUuid, fileName, cellrangerVersion) {
+    const s3 = new AWS.S3({ apiVersion: '2006-03-01', signatureVersion: 'v4' });
+
+    const params = {
+      Bucket: this.sampleFilesBucketName,
+      Key: `${projectUuid}/${sampleUuid}/${fileName}`,
+      Expires: 360,
+    };
+
+    if (cellrangerVersion) {
+      params.Metadata = {
+        cellranger_version: cellrangerVersion,
```

Comment:
Is this a new metadata or do we have this already?

Reply:
`cellranger_version` is a special header we need to send that, if we follow a certain pattern, is then added as metadata to an s3 file, it is information used by gem2s.

Before, we sent this when uploading the samples in the ui but now we need to send it when generating the url.